src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -6827,23 +6827,41 @@ def delete_model_version_artifact_link(
             session.delete(model_version_artifact_link)
             session.commit()
 
-    def delete_all_model_version_artifact_link(
+    def delete_all_model_version_artifact_links(
         self,
         model_version_id: UUID,
+        only_links: bool = True,
     ) -> None:
         """Deletes all model version to artifact links.
 
         Args:
             model_version_id: ID of the model version containing the link.
+            only_links: Whether to only delete the link to the artifact.
         """
         with Session(self.engine) as session:
-            model_version = self.get_model_version(model_version_id)
+            if not only_links:
```

Comment:
The versions, not the artifacts. Artifacts can be cleaned with pruning from another PR.

Side thing (more related to second PR): since we allow pruning without deletion of actual data it will result in orphaned data in certain cases, which cannot be deleted by regular means, since no entries are present in DB anymore. This might be a thing to consider as well.